docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,223 @@
+# Migration from v1 to v2
+
+Make ensure you understand the [differences between Kubebuilder v1 and v2](./v1vsv2.md)
+before continuing
+
+Please ensure you have followed the [installation guide](/quick-start.md#installation)
+to install the required components.
+
+The recommended way to migrate a v1 project is to create a new v2 project and
+copy over the API and the reconciliation code. The conversion will end up with a
+project that looks like a native v2 project. However, in some cases, it's
+possible to do a in-place upgrade. i.e. reuse the v1 project layout and only
+upgrade controller-runtime and controller-tools.  
+
+Let's take the [example v1 project][v1-project] and migrate it to Kubebuilder
+v2. At the end, we should have something that looks like the
+[example v2 project][v2-project].
+
+## Preparation
+
+What are the group, version, kind and domain? We'll need to figure out what the
```

Comment:
No need for the first question -- you can just say `First, we'll need to figure out ...` without the `What are the ...`:

```suggestion
First, we'll need to figure out what the
```